generated/nidaqmx/task.py

```diff
@@ -760,6 +760,46 @@ def read(self, number_of_samples_per_channel=NUM_SAMPLES_UNSET,
 
         return data.tolist()
 
+    def add_arguments_if_necessary(self, callback_function, expected_number_of_arguments):
+        if (callback_function.__code__.co_argcount < expected_number_of_arguments):
+            print("Not enough arguments! Adding arguments")
+            if ("every_n_samples_event_type" in parameters):
+                result = self.n_samples_event_wrapper(parameters, callback_function)
+            elif ("signal_type" in parameters):
+                result = self.register_signal_event_wrapper(parameters, callback_function)
+            elif ("status" in parameters):
+                result = self.register_done_event_wrapper(parameters, callback_function)
+        else:
+            result = callback_function
+        return result
+
+    def n_samples_event_wrapper(self, parameters, callback_function):
+        if ("task_handle" and "callback_data" not in parameters):
+            result = lambda task_handle, every_n_samples_event_type, number_of_samples, callback_data: callback_function(every_n_samples_event_type, number_of_samples)
+        elif ("task_handle" not in parameters):
+            result = lambda task_handle, every_n_samples_event_type, number_of_samples, callback_data: callback_function(every_n_samples_event_type, number_of_samples, callback_data)
+        elif ("callback_data" not in parameters):
+            result = lambda task_handle, every_n_samples_event_type, number_of_samples, callback_data: callback_function(task_handle, every_n_samples_event_type, number_of_samples)
+        return result
+
+    def register_signal_event_wrapper(self, parameters, callback_function):
+        if ("task_handle" and "callback_data" not in parameters):
+            result = lambda task_handle, signal_type, callback_data: callback_function(signal_type)
+        elif ("task_handle" not in parameters):
+            result = lambda task_handle, signal_type, callback_data: callback_function(signal_type, callback_data)
+        elif ("callback_data" not in parameters):
```

Comment:
Here's an example of what I mean:

```
def register_done_event(self, callback_method, callback_data = None):
   ...
```

It's not critical because Python makes it easy to bind a closure or to pass in a member function as the callback, but it's a potential nice-to-have and it would avoid removing parameters from the signature.